### PR TITLE
GG-24425: DR: Sender workers doesn't stop on deactivation.

### PR DESCRIPTION
**Pull request by AMashenkov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java [hunk 1]
```diff
@@ -48,6 +48,9 @@
     /** Flag indicating if the cluster in read-only mode. */
     private final boolean readOnly;
 
+    /** Last cluster activation time. */
+    private final long activationTime;
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java]: Looks like this is not activation time, but time when state changed for the last time. May be we should name it stateChangeTime or something like this.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java [hunk 2]
```diff
@@ -129,7 +135,7 @@ static DiscoveryDataClusterState createTransitionState(
             baselineTopology,
             transitionReqId,
             transitionTopVer,
-            transitionNodes
+            activationTime, transitionNodes
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java]: transitionNodes should be on the next line.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java [hunk 3]
```diff
@@ -148,11 +155,12 @@ private DiscoveryDataClusterState(
         @Nullable BaselineTopology baselineTopology,
         @Nullable UUID transitionReqId,
         @Nullable AffinityTopologyVersion transitionTopVer,
-        @Nullable Set<UUID> transitionNodes
+        long activationTime, @Nullable Set<UUID> transitionNodes
```
Comment (korlov42) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java]: transitionNodes should be on the next line.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/datastreamer/MultipleDataStreamersOnClientTest.java [hunk 4]
```diff
@@ -1,253 +1,253 @@
-/*
```
Comment (korlov42) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/datastreamer/MultipleDataStreamersOnClientTest.java]: could you please highlight what had changed here?

Comment (AMashenkov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/datastreamer/MultipleDataStreamersOnClientTest.java]: Contents have differences only in line separators

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java (hunk 1 above)
Comment (korlov42) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/DiscoveryDataClusterState.java]: but it’s better to correct the condition on GridClusterStateProcessor.java:642. We should get time from message only if msg.activate() is true and state.active() is false, if I understood the idea correctly

**Pull request closed without merging**